pkg/cmd/cli/cmd/expose.go

```diff
@@ -56,7 +56,13 @@ func NewCmdExpose(fullName string, f *clientcmd.Factory, out io.Writer) *cobra.C
 		cmdutil.CheckErr(err)
 		defRun(cmd, args)
 	}
-	cmd.Flags().String("hostname", "", "Set a hostname for the new route")
+	cmd.Flags().String("hostname", "", "Set a hostname for the new route. Optional.")
+	cmd.Flags().String("path", "", "Path that the router watches to route traffic to the service. Optional.")
+	cmd.Flags().String("tls-termination", "", "Indicate a termination type for the generated route. If not set, any termination config will be ignored. Supports 'edge', 'passthrough', or 'reencrypt'.")
+	cmd.Flags().String("cert-file", "", "Provide certificate contents for a secure route. Used by edge and re-encryption terminated routes.")
```

Comment:
> Meet the intersection of flags and generators. Currently, flags are copied verbatim, by name, into a magic map that is passed to a "generate" method that can call any number of different generators, each of which must agree on the meaning of every single flag. We have to make sure that match not only the field, but also the meanings and types of every compatible generator upstream.

I don't think this problem starts from the generators. Are we sure that a `--route-client-certificate` flag will always be used for routes? If not, then why did we name it like that in the first place? But even in the event of using `--route-client-certificate` for things other than routes then a generator won't stand on your way (why they are flexible minus pretty:) ). That being said, I still prefer `--cert-file`.